common/file.h

```diff
@@ -1,152 +0,0 @@
-/*******************************************************************************
- * Thunderbolt(TM) tbtadm tool
- * This code is distributed under the following BSD-style license:
- *
- * Copyright(c) 2017 Intel Corporation.
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *
- *     * Redistributions of source code must retain the above copyright notice,
- *       this list of conditions and the following disclaimer.
- *     * Redistributions in binary form must reproduce the above copyright
- *       notice, this list of conditions and the following disclaimer in the
- *       documentation and/or other materials provided with the distribution.
- *     * Neither the name of Intel Corporation nor the names of its contributors
- *       may be used to endorse or promote products derived from this software
- *       without specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
- * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
- ******************************************************************************/
-
-#pragma once
-
-#include <string>
-
-#include <fcntl.h> // for O_RDONLY, O_WRONLY
-
-#ifdef USE_STD_FS
-#include <experimental/filesystem>
-namespace fs = std::experimental::filesystem;
-#else
-#include <boost/filesystem.hpp>
-namespace fs = boost::filesystem;
-#endif
-
-namespace tbtadm
-{
-/**
- * @brief This class wraps-around POSIX file interface for C++ style usage
- *
- * The class encapsulates the actions to be type-safe, translates errors to
- * exceptions so they will not get ignored and implements RAII-style handling of
- * the file.
- *
- * The operations work on the file as whole, e.g. reading the whole file
- * together, as this is the only use-case we need right now.
- *
- * The assumption here is that the class is used only for sysfs files (e.g. non-
- * seekable).
- *
- * The reason all this is done instead of simply using the already existing C++
```

Comment:
Have you read this paragraph of the documentation?
Did you found a better solution to get the real `errno` from sysfs in case of an error?

Reply:
I can add some more error handling, looks like I can use errno also with fstream operations.
https://stackoverflow.com/questions/17337602/how-to-get-error-message-when-ifstream-open-fails

Reply:
Yeah, the fun begins when you realize that there is no guarantee that you get the `errno` for the read failure, there can be other internal stream actions done later that overwrites it with the error resulted from those actions... So you maybe can test it and claim Works For Me (tm) but there is no guarantee at all that it will always report the correct error.
This is exactly why I resorted to wrapping the posix calls this way, because this is the only way that promises getting the correct and actual error.

(BTW, this is mentioned in the page you linked to... Read the comments too, not only the answers. And it also mentioned in the documentation of the `File` class.)